static/app/components/events/interfaces/crashContent/exception/sourceMapDebug.tsx

```diff
@@ -0,0 +1,244 @@
+            error.data.matchedSourcemapPath
+          ),
+          docsLink:
+            platform === 'javascript'
+              ? 'https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/#verify-artifact-names-match-stack-trace-frames'
+              : `https://docs.sentry.io/platforms/javascript/guides/${platformDocsMap[platform]}/sourcemaps/troubleshooting_js/#verify-artifact-names-match-stack-trace-frames`,
+        },
+      ];
+    case SourceMapProcessingIssueType.UNKNOWN_ERROR:
+      return [
+        {
+          title: t('UNKNOWN_ERROR'),
+        },
+      ];
+    case SourceMapProcessingIssueType.MISSING_USER_AGENT:
+      return [
+        {
+          title: t('MISSING_USER_AGENT'),
+        },
+      ];
+    case SourceMapProcessingIssueType.MISSING_SOURCEMAPS:
+      return [
+        {
+          title: t('MISSING_SOURCEMAPS'),
+        },
+      ];
+    case SourceMapProcessingIssueType.URL_NOT_VALID:
+      return [
+        {
+          title: t('URL_NOT_VALID'),
+        },
+      ];
+    default:
+      return [];
+  }
+}
+
+/**
+ * Kinda making this reuseable since we have this pattern in a few places
+ */
+function ExpandableErrorList({
+  title,
+  children,
+  docsLink,
+}: {
+  title: React.ReactNode;
+  children?: React.ReactNode;
+  docsLink?: React.ReactNode;
+}) {
+  const [expanded, setExpanded] = useState(false);
+  return (
+    <List symbol="bullet">
+      <StyledListItem>
+        <ErrorTitleFlex>
+          <ErrorTitleFlex>
+            <strong>{title}</strong>
+            {children && (
+              <ToggleButton
+                priority="link"
+                size="zero"
+                onClick={() => setExpanded(!expanded)}
+              >
+                {expanded ? t('Collapse') : t('Expand')}
+              </ToggleButton>
+            )}
+          </ErrorTitleFlex>
+          {docsLink}
+        </ErrorTitleFlex>
+
+        {expanded && <div>{children}</div>}
+      </StyledListItem>
+    </List>
+  );
+}
+
+function combineErrors(
+  response: Array<SourceMapDebugResponse | undefined | null>,
+  platform: PlatformType
+) {
+  const combinedErrors = uniqBy(
+    response
+      .map(res => res?.errors)
+      .flat()
+      .filter(defined),
+    error => error?.type
+  );
+  const errors = combinedErrors.map(error => getErrorMessage(error, platform)).flat();
+
+  return errors;
+}
+
+interface SourcemapDebugProps {
+  /**
+   * A subset of the total error frames to validate sourcemaps
+   */
+  debugFrames: StacktraceFilenameQuery[];
+  platform: PlatformType;
+}
+
+export function SourceMapDebug({debugFrames, platform}: SourcemapDebugProps) {
+  const organization = useOrganization();
+  const [firstFrame, secondFrame, thirdFrame] = debugFrames;
+  const hasFeature = organization?.features?.includes('fix-source-map-cta');
+  const queryOptions = {enabled: hasFeature};
+  const {data: firstData} = useSourceMapDebug(firstFrame?.query, queryOptions);
```

Comment:
Instead of 0-3 useQuery calls, would it make more sense to use [useQueries](https://tanstack.com/query/v4/docs/react/reference/useQueries)?